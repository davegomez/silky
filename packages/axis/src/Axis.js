// @flow
import React from 'react';
import type { CSS } from '../../utils/types';
import { Group } from '@silky/group';

type Props = {
  className?: string,
  css?: CSS,
  position: string,
};

export default function Text({ className = '', css, position }: Props) {
  return (
    <Group className={`silky-axis axis-${position} ${className}`} css={css}>
      {/* Test */}
    </Group>
  );
}
