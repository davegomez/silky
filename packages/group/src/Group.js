// @flow
import React from 'react';
import { getTransform } from '../../utils';
import type { CSS, Transform } from '../../utils/types';

type Props = {
  children: any,
  className?: string,
  css?: CSS,
  transform?: Transform,
};

export default function Group({
  children,
  className = '',
  css,
  transform,
  ...restProps
}: Props) {
  return (
    <g
      className={`silky-group ${className}`}
      css={css}
      transform={getTransform(transform)}
      {...restProps}
    >
      {children}
    </g>
  );
}
