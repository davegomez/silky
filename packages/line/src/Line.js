// @flow
import React from 'react';
import type { CSS } from '../../utils/types';

type Props = {
  className?: string,
  css?: CSS,
  x1: number | string,
  x2?: number | string,
  y1: number | string,
  y2?: number | string,
  pathLength?: number,
};

export default function Line({
  className = '',
  css,
  x1,
  x2,
  y1,
  y2,
  pathLength,
  ...restProps
}: Props) {
  return (
    <line
      className={`silky-line ${className}`}
      css={css}
      x1={x1}
      x2={x2}
      y1={y1}
      y2={y2}
      pathLength={pathLength}
      {...restProps}
    />
  );
}
