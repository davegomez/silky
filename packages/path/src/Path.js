// @flow
import React from 'react';
import type { CSS } from '../../utils/types';

type Props = {
  className?: string,
  css?: CSS,
  d: string,
  pathLength?: number,
};

export default function Path({
  className = '',
  css,
  d,
  pathLength,
  ...restProps
}: Props) {
  return (
    <path
      className={`silky-path ${className}`}
      css={css}
      d={d}
      pathLength={pathLength}
      {...restProps}
    />
  );
}
