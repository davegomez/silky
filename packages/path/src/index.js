export { default as Path } from './Path';
