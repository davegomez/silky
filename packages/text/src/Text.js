// @flow
import React from 'react';
import { getTransform } from '../../utils';
import type { CSS, Transform } from '../../utils/types';

type Props = {
  children: any,
  className?: string,
  css?: CSS,
  dx?: number | string,
  dy?: number | string,
  lengthAdjust?: 'spacing' | 'spacingAndGlyphs',
  transform?: Transform,
  textLength?: number | string,
  x?: number | string,
  y?: number | string,
};

export default function Text({
  children,
  className = '',
  css,
  dx,
  dy,
  lengthAdjust,
  transform,
  textLength,
  x,
  y,
  ...restProps
}: Props) {
  return (
    <text
      className={`silky-text ${className}`}
      css={css}
      dx={dx}
      dy={dy}
      lengthAdjust={lengthAdjust}
      textLength={textLength}
      transform={getTransform(transform)}
      x={x}
      y={y}
      {...restProps}
    >
      {children}
    </text>
  );
}
