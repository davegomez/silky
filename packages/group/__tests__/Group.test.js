import React from 'react';
import serializer, { matchers } from 'jest-emotion';
import { render, cleanup } from 'react-testing-library';
import { css } from '@emotion/core';
import { Group } from '../src';

expect.addSnapshotSerializer(serializer);
expect.extend(matchers);
afterEach(cleanup);

test('G', () => {
  const { container } = render(
    <svg>
      <Group>foo</Group>
    </svg>
  );

  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.classList.contains('silky-group')).toBe(true);
  expect(group.innerHTML).toBe('foo');
});

test('Group with custom class', () => {
  const { container } = render(
    <svg>
      <Group className="foo">foo</Group>
    </svg>
  );

  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.classList.contains('silky-group')).toBe(true);
  expect(group.classList.contains('foo')).toBe(true);
  expect(group.innerHTML).toBe('foo');
});

test('Group with Emotion', () => {
  const styles = css`
    color: black;
  `;
  const { container } = render(
    <svg>
      <Group css={styles}>foo</Group>
    </svg>
  );

  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.classList.contains('silky-group')).toBe(true);
  expect(group).toHaveStyleRule('color', 'black');
  expect(group.innerHTML).toBe('foo');
});

test('Group width transform translate', () => {
  const { container } = render(
    <svg>
      <Group transform={{ left: 10, top: 10 }}>foo</Group>
    </svg>
  );

  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.classList.contains('silky-group')).toBe(true);
  expect(group.getAttribute('transform')).toEqual('translate(10, 10) ');
  expect(group.innerHTML).toBe('foo');
});

test('Group width transform rotate', () => {
  const { container } = render(
    <svg>
      <Group transform={{ rotation: '10 20, 30' }}>foo</Group>
    </svg>
  );

  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.classList.contains('silky-group')).toBe(true);
  expect(group.getAttribute('transform')).toEqual(' rotate(10 20, 30)');
  expect(group.innerHTML).toBe('foo');
});

test('Group width transform translate and rotate', () => {
  const { container } = render(
    <svg>
      <Group transform={{ left: 10, top: 10, rotation: 90 }}>foo</Group>
    </svg>
  );

  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.classList.contains('silky-group')).toBe(true);
  expect(group.getAttribute('transform')).toEqual(
    'translate(10, 10) rotate(90)'
  );
  expect(group.innerHTML).toBe('foo');
});
