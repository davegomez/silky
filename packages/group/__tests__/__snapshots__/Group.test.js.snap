// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`G 1`] = `
<g
  class="silky-group "
>
  foo
</g>
`;

exports[`Group with Emotion 1`] = `
.emotion-0 {
  color: black;
}

<g
  class="silky-group emotion-0"
>
  foo
</g>
`;

exports[`Group with custom class 1`] = `
<g
  class="silky-group foo"
>
  foo
</g>
`;
